test/e2e/apimachinery/execwithoptions_stress.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apimachinery
+
+import (
+	"bytes"
+	"context"
+	"path"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"k8s.io/apimachinery/pkg/labels"
+	commonutils "k8s.io/kubernetes/test/e2e/common"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2edebug "k8s.io/kubernetes/test/e2e/framework/debug"
+	e2ekubectl "k8s.io/kubernetes/test/e2e/framework/kubectl"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	e2etestfiles "k8s.io/kubernetes/test/e2e/framework/testfiles"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = SIGDescribe("execWithOptions stress", func() {
+
+	f := framework.NewDefaultFramework("execwithoptions-stress")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	var podYaml string
+
+	ginkgo.BeforeEach(func(ctx context.Context) {
+		ginkgo.By("creating a busybox pod")
+		data, err := e2etestfiles.Read(path.Join("test/e2e/testing-manifests/kubectl/busybox-pod.yaml.in"))
+		if err != nil {
+			framework.Fail(err.Error(), 1)
+		}
+		podYaml = commonutils.SubstituteImageName(string(data))
+		e2ekubectl.RunKubectlOrDieInput(f.Namespace.Name, podYaml, "create", "-f", "-")
+
+		ginkgo.By("waiting for busybox's availability")
+		_, err = e2epod.WaitForPodsWithLabelRunningReady(ctx, f.ClientSet, f.Namespace.Name,
+			labels.Set{"app": "busybox1"}.AsSelector(), 1 /* one replica */, 100*time.Second)
+		if err != nil {
+			e2edebug.DumpAllNamespaceInfo(ctx, f.ClientSet, f.Namespace.Name)
+			e2ekubectl.LogFailedContainers(ctx, f.ClientSet, f.Namespace.Name, framework.Logf)
+			framework.Failf("unable to wait for all pods to be running and ready: %v", err)
+		}
+	})
+
+	ginkgo.It("execWithOptions stress test", func(ctx context.Context) {
+		dat_bytes := []byte("")
+		for {
```

Comment:
For a test that (eventually) will run in the CI, we have to have some termination criteria. We could run for a certain number of iterations and then choose the number so that we don't exceed a reasonable overall runtime, but a better choice might be to make it time-based directly, i.e. run it for a few minutes.

The other aspect is that this test currently doesn't generate much stress for the apiserver because it runs everything sequentially. It's better to create several workers that concurrently do this check.

Reply:
Yeah, I know, but just I wanted to reproduce the problem in your issue first..!
Thank you for detailed explanation though.!

Reply:
I tried in many ways using waitgroup, and channels with go routines, but I could not implement "running the e2e test for a specific duration"...
Could you give me some example that does something similar?

Reply:
I don't know if it is a good idea to have a duration in this way, but just set with go func() that sets the value of 'isTimeout' as true after sleep some time, and then inside the for loop checked whether the variable was set or not.

Reply:
See https://github.com/hj-johannes-lee/kubernetes/pull/1

Reply:
Thank you for this advice and improvement.!!